File: jsol-compiler-py/0110-config-parser.py
import math
import functools
from jsol_core import JSOL

# @JSOL v0.2.97 - Targets Configuration Normalizer (generic, target-agnostic)
def mNormalizeTargetsConfig(mRawConfig): 

  aTargetIds = list(mBackendRegistry.keys());
  mResult = JSOL.dict();

  i = 0;
  while i < len(aTargetIds): 

    saTargetId = aTargetIds[i];
    mEmptySub = JSOL.dict();
    mResult[saTargetId] = JSOL.dict("default",  "",  "targets",  mEmptySub);

    i = i + 1;


  if mRawConfig == None: 

    return mResult;


  i = 0;
  while i < len(aTargetIds): 

    saTargetId = aTargetIds[i];
    if ( saTargetId in mRawConfig) == True: 

      mSub = mRawConfig[saTargetId];
      if ( "default" in mSub) == True: 

        mResult[saTargetId]["default"] = mSub["default"];


      if ( "targets" in mSub) == True: 

        mResult[saTargetId]["targets"] = mSub["targets"];




    i = i + 1;


  if ( "default" in mRawConfig) == True and ( "targets" in mRawConfig) == True: 

    i = 0;
    while i < len(aTargetIds): 

      saTargetId = aTargetIds[i];
      mResult[saTargetId]["default"] = mRawConfig["default"];
      mResult[saTargetId]["targets"] = mRawConfig["targets"];

      i = i + 1;




  return mResult;

File: jsol-compiler-ts/0110-config-parser.ts
declare var JSOL: any;
declare var Rgx: any;
declare var Str: any;
declare var Arr: any;
declare var Bool: any;
declare var Cast: any;

// @JSOL v0.2.97 - Targets Configuration Normalizer (generic, target-agnostic)
const $mNormalizeTargetsConfig = function($mRawConfig: any): Record<string, any> {
  const $aTargetIds: any[] = Object.keys($mBackendRegistry);
    const $mResult: Record<string, any> = JSOL.dict();

    for (let $i = 0; $i < $aTargetIds.length; $i = $i + 1) {
    const $saTargetId: string = $aTargetIds[$i];
        const $mEmptySub: Record<string, any> = JSOL.dict();
        $mResult[$saTargetId] = JSOL.dict("default",  "",  "targets",  $mEmptySub);
  }
  if ($mRawConfig === null) {
    return $mResult;
  }
  for (let $i = 0; $i < $aTargetIds.length; $i = $i + 1) {
    const $saTargetId: string = $aTargetIds[$i];
        if (Object.prototype.hasOwnProperty.call($mRawConfig,  $saTargetId) === true) {
      const $mSub: Record<string, any> = $mRawConfig[$saTargetId];
            if (Object.prototype.hasOwnProperty.call($mSub,  "default") === true) {
        $mResult[$saTargetId]["default"] = $mSub["default"];
      }
      if (Object.prototype.hasOwnProperty.call($mSub,  "targets") === true) {
        $mResult[$saTargetId]["targets"] = $mSub["targets"];
      }
    }
  }
  if (Object.prototype.hasOwnProperty.call($mRawConfig,  "default") === true && Object.prototype.hasOwnProperty.call($mRawConfig,  "targets") === true) {
    for (let $i = 0; $i < $aTargetIds.length; $i = $i + 1) {
      const $saTargetId: string = $aTargetIds[$i];
            $mResult[$saTargetId]["default"] = $mRawConfig["default"];
            $mResult[$saTargetId]["targets"] = $mRawConfig["targets"];
    }
  }
  return $mResult;
};
